ni_measurementlink_generator/ni_measurementlink_generator/templates/measurement.serviceconfig.mako

```diff
@@ -7,7 +7,12 @@
       "serviceClass": "${service_class}",
       "descriptionUrl": "${description_url}",
       "providedInterfaces": [ "ni.measurementlink.measurement.v1.MeasurementService" ],
-      "path": "start.bat"
+      "path": "start.bat",
+      "annotations": {
+        "ni/service.description": "${annotations_description}",
+        "ni/service.collection": "${collection}",
+        "ni/service.tags": ${tags}
```

Comment:
This highlights the fact that we are generating JSON with a mako template. This is a bad practice because there are a lot of cases it doesn't handle correctly. For example, strings that contain quotes or whitespace should use JSON escape sequences.

It would be better to build a nested dict and use `json.dumps` to convert it to JSON:

```
<% 
    service_config = {
        "services": [
            {
                "displayName": display_name,
                "serviceClass": service_class,
                ...
            }
        ]
    }
%>\
${json.dumps(service_config, indent=2)}
```

Dict preserves key order in recent Python versions, so I don't think we need OrderedDict to have a stable key order.

You could add `separators=` if needed but I don't know if we care.

Reply:
I don't think we have any problem with the whitespace, since the description already has them. Tbh I don't think we are expecting any complex string input from the customer on these specific annotations. If this is important for the customer we can add this.

Reply:
I don't want to maintain a template that generates JSON in this way.

I posted a separate PR to fix this: https://github.com/ni/measurementlink-python/pull/314

Reply:
hey Brad,  sorry I missed you posted the whole solution here.